### PR TITLE
Obit r632

### PR DESCRIPTION
**Pull request by mauch**
This primarily fixes the missing facet issue reported in https://skaafrica.atlassian.net/browse/SPR1-566 but also contains updates to deal with other fixes in R632. I've tried to explain whats going on in the commit messages - but please ask if anything too Obitty doesn't make sense :)

### CODE REVIEW

File: katacomb/katacomb/img_facade.py [hunk 1]
```diff
@@ -426,9 +426,11 @@ def GetPlane(self, array=None, plane=[1, 1, 1, 1, 1]):
                    "from image '%s'" % (plane, self.name))
 
         try:
-            if array is None:
-                array = Image.PGetFArray(self.img)
             self.img.GetPlane(array, plane, self._err)
+            if array is None:
+                # Plane is retrieved to self.img.FArray
+                # Return a deep copy.
+                array = FArray.PCopy(self.img.FArray, self._err)
```
Comment (bmerry) [inline katacomb/katacomb/img_facade.py:433]: I don't have the big picture of the error handling, but do you need to check whether GetPlane put an error into `self._err` before doing this step (which presumably *also* updates `self._err` with error information)?

File: katacomb/katacomb/img_facade.py [hunk 2]
```diff
@@ -448,20 +450,22 @@ def PutPlane(self, array, plane=[1, 1, 1, 1, 1]):
...
+    def FitMF(self, out_path, **kwargs):
         """
         Fit spectrum to each pixel of an ImageMF.
 
-        Uses ImageMF.PFitSpec and takes kwargs (from ImageMF.PFitSpec docstring).
+        Uses ImageMF.PFitSpec2 and passes kwargs (from ImageMF.PFitSpec2 docstring).
 
         Parameters
         ----------
+        out_path : :class:AIPSPath
```
Comment (bmerry) [inline katacomb/katacomb/img_facade.py:461]:
```suggestion
        out_path : :class:`AIPSPath`
```

File: katacomb/katacomb/img_facade.py [hunk 3]
```diff
@@ -471,12 +475,15 @@ def FitMF(self, **kwargs):
...
-            handle_obit_err(err_msg)
+            with img_factory(aips_path=out_path, mode='rw') as out_img:
+                try:
+                    # Copy the structure of self.img to out_img
+                    self.img.Clone(out_img.img, self._err)
+                    # Use an imgMF pointer for PFitSpec2
+                    ImageMF.PFitSpec2(self.imgMF, out_img.imgMF, self._err, **kwargs)
+                except Exception:
+                    raise Exception(err_msg)
+                handle_obit_err(err_msg)
```
Comment (bmerry) [inline katacomb/katacomb/img_facade.py:486]: Any reason `self._err` isn't being passed to `handle_obit_err` here?

**Review by bmerry — CHANGES_REQUESTED** [review]
Frankly I don't understand any of it (not knowing anything about Obit), but I've pointed out a couple of places that look a touch odd to me.

File: katacomb/katacomb/img_facade.py (hunk 1 above)
Comment (mauch) [reply katacomb/katacomb/img_facade.py:433]: The way it works is that if a task (i.e. GetPlane) puts an error flag into self._err all future task calls using the same `_err` (i.e. FArray.PCopy) will do nothing.
The subsequent `handle_obit_error` call will then log the error from GetPlane once it is reached if GetPlane flagged something into self._err.

File: katacomb/katacomb/img_facade.py (hunk 3 above)
Comment (mauch) [reply katacomb/katacomb/img_facade.py:486]: `self._err` comes from the context in which the pipeline is run, in this case an `ObitContext` see: https://github.com/ska-sa/katsdpcontim/blob/master/katacomb/katacomb/obit_context.py#L20 . `handle_obit_error` will default to the error handler defined in the current context if it isn't set (`self._err` is just defined by retrieving the one from the current context and is passed to any Obit task that requires it).

**Review by bmerry — APPROVED** [review]

**Comment by bmerry** [comment]
Ok, if you're happy with the error handling then I'll trust your experience with Obit.

**Pull request merged**